libfreerdp/core/listener.c

```diff
@@ -258,16 +258,20 @@ static BOOL freerdp_listener_get_fds(freerdp_listener* instance, void** rfds, in
 int freerdp_listener_get_event_handles(freerdp_listener* instance, HANDLE* events, DWORD* nCount)
 {
 	int index;
+	DWORD count = *nCount;
 	rdpListener* listener = (rdpListener*) instance->listener;
 
 	if (listener->num_sockfds < 1)
 		return -1;
 
+	if (listener->num_sockfds > count)
+		return -1;
+
 	for (index = 0; index < listener->num_sockfds; index++)
 	{
-		events[*nCount] = listener->events[index];
-		(*nCount)++;
+		events[index] = listener->events[index];
 	}
+	*nCount = listener->num_sockfds;
 
 	return 0;
```

Comment:
@akallabeth would be nice if we could make this function behave like the other _get_handles function. So what do you think about returning 0 on error and the number of events otherwise?